redash/query_runner/query_results.py

```diff
@@ -60,21 +65,29 @@ def _load_query(user, query_id):
     return query
 
 
-def create_tables_from_query_ids(user, connection, query_ids):
-    for query_id in set(query_ids):
+def get_query_results(user, query_id, bring_from_cache):
         query = _load_query(user, query_id)
+        if bring_from_cache and query.latest_query_data_id is not None:
+            results = query.latest_query_data.data
+        else: 
+            results, error = query.data_source.query_runner.run_query(query.query_text, user)
+            if error:
+                raise Exception(
+                    "Failed loading results for query id {}.".format(query.id))
+        return json.loads(results)
 
-        results, error = query.data_source.query_runner.run_query(
-            query.query_text, user)
 
-        if error:
-            raise Exception(
-                "Failed loading results for query id {}.".format(query.id))
-
-        results = json.loads(results)
+def create_tables_from_query_ids(user, connection, query_ids, cached_query_ids=[]):
+    for query_id in set(query_ids):
+        results = get_query_results(user, query_id, False)
         table_name = 'query_{query_id}'.format(query_id=query_id)
         create_table(connection, table_name, results)
 
+    for query_id in set(cached_query_ids):
```

Comment:
It's worth loading the cached results first.

Reply:
I hope you mean to run this for loop first, 
That's what I changed in my last commit